src/wallet/rpcwallet.cpp

```diff
@@ -244,6 +244,107 @@ UniValue getaddressinfo(const JSONRPCRequest& request)
     return ret;
 }
 
+UniValue getaddressesbylabel(const JSONRPCRequest& request)
+{
+    CWallet* const pwallet = pwalletMain;
+    if (!pwallet) {
+        return NullUniValue;
+    }
+
+    if (request.fHelp || request.params.size() != 1)
+        throw std::runtime_error(
+            "getaddressesbylabel \"label\"\n"
+            "\nReturns the list of addresses assigned the specified label.\n"
+
+            "\nArguments:\n"
+            "1. \"label\"  (string, required) The label.\n"
+
+            "\nResult:\n"
+            "{ (json object with addresses as keys)\n"
+            "  \"address\": { (json object with information about address)\n"
+            "    \"purpose\": \"string\" (string)  Purpose of address (\"send\" for sending address, \"receive\" for receiving address)\n"
+            "  },...\n"
+            "}\n"
+
+            "\nExamples:\n"
+            + HelpExampleCli("getaddressesbylabel", "\"tabby\"")
+            + HelpExampleRpc("getaddressesbylabel", "\"tabby\"")
+        );
+
+    LOCK(pwallet->cs_wallet);
+
+    std::string label = LabelFromValue(request.params[0]);
+
+    // Find all addresses that have the given label
+    UniValue ret(UniValue::VOBJ);
+    for (const std::pair<CTxDestination, AddressBook::CAddressBookData>& item : pwallet->mapAddressBook) {
+        if (item.second.name == label) {
+            ret.pushKV(EncodeDestination(item.first), AddressBookDataToJSON(item.second, false));
```

Comment:
We will have to review this flow, `EncodeDestination` needs a boolean to know whether the `CKeyID` needs to be formatted into a staking address or a regular one.

As it is now, it will wrongly return the regular address format for staking addresses.

Reply:
fixed, and added a functional test to verify the intended behavior

Reply:
functional test:
https://github.com/PIVX-Project/PIVX/pull/1731/commits/566e2019edb52f8e4031556d6947d68d7eb3370d#diff-feedc65baf14793d567149a8b7e256a1R157-R168